### PR TITLE
Pl 158 summary/key components refactor

### PR DESCRIPTION
**Pull request by matteovisotto**
DONE:
Created unique AccordionItem class and subclasses for each specific item, avoiding code repetition and overriding only if necessary
All function are async/await and then has been removed

### CODE REVIEW

File: src/components/SummaryKeyComponents/AbstractSummaryAccordionItem.js [hunk 1]
```diff
@@ -1,43 +1,24 @@
...
-  /**
-   * Set loading constant as a text
-   */
-  setLoading = () => {
-    this.components.abstractSummaryText.innerText = LOADING_TEXT;
-  };
-
-  /**
-   * Clear the content setting the text to empty string
+   * Set the text af the abstract
```
Comment (NinaGnjidic) [inline src/components/SummaryKeyComponents/AbstractSummaryAccordionItem.js:17]: Smally typo

**Review by NinaGnjidic — COMMENTED** [review]
Wow Matteo this is awesome, all one liners 😍 , reading your code made my day so far, +1

**Comment by dariomesic** [comment]
+1 by me, your code is next level for sure so i can't add anything

**Comment by tonesija** [comment]
+1 nice job

**Pull request merged**